app/assets/javascripts/application.js

```diff
@@ -544,6 +544,43 @@ function closeModalSpinner(spinnerTarget, modalTarget, callback) {
     $(modalTarget).modal('hide');
 }
 
+// handles showing/hiding main message_modal and cleaning up state on full & partial page renders
+function showMessageModal(notice=null, alert=null) {
+    console.log('calling showMessageModal');
+    // close any open modals
+    if (OPEN_MODAL) {
+        var modalTarget = $('#' + OPEN_MODAL);
+        var modalData = modalTarget.data('bs.modal');
+        if ( typeof modalData !== 'undefined' && modalData.isShown) {
+            modalTarget.modal("hide");
+        }
+    }
+
+    var noticeElement = $('#notice-content');
+    var alertElement = $('#alert-content');
+    if (notice) {
+        noticeElement.html(notice);
+    } else {
+        noticeElement.empty();
+    }
+    if (alert) {
+        alertElement.html("<strong>" + alert + "</strong>");
+    } else {
+        alertElement.empty();
+    }
+
+    if (notice || alert) {
+        $("#message_modal").modal("show");
+    }
+
+    // don't timeout alert messages
+    if (alert) {
+        setTimeout(function() {
+            $("#message_modal").modal("hide");
+        }, 3000);
+    }
+}
```

Comment:
Does this code mean "if there is an alert, hide the message modal after 3 seconds"?  If so:

```suggestion
    // timeout alert messages
    if (alert) {
        setTimeout(function() {
            $("#message_modal").modal("hide");
        }, 3000);
    }
}
```

Elsewhere, I see `if (<%= alert.nil? %>) {` with another copy of this code snippet, which hints that maybe `if (!alert) {` is intended here.

Either way, either the code or the comment seems incorrect.

Reply:
You're correct - if there is an alert message present, it should not call `setTimeout()`.  I'll update that as well.